src/goose/brokers/redis/cron/registry.clj

```diff
@@ -0,0 +1,64 @@
+(ns goose.brokers.redis.cron.registry
+  (:require [goose.defaults :as d]
+            [goose.brokers.redis.commands :as redis-cmds]
+            [taoensso.carmine :as car]
+            [goose.job :as j]
+            [goose.cron.parsing :as cron-parsing]
+            [goose.utils :as u]))
+
+(defn registry-entry [cron-schedule job-description]
+  {:id              (str (random-uuid))
+   :cron-schedule   cron-schedule
+   :job-description job-description})
+
+(defn register-cron
+  [conn cron-schedule job-description]
+  (let [entry             (registry-entry cron-schedule job-description)
+        next-run-epoch-ms (cron-parsing/next-run-epoch-ms cron-schedule)]
+    (redis-cmds/enqueue-sorted-set conn
+                                   d/prefixed-cron-schedules-queue
+                                   next-run-epoch-ms
+                                   entry)
+    entry))
+
+(defn set-due-time [cron-entry next-run-epoch-ms]
+  (car/zadd d/prefixed-cron-schedules-queue "XX" next-run-epoch-ms cron-entry))
+
+(defn- enqueue-jobs-to-ready-on-priority [jobs]
+  (doseq [[queue-key grouped-jobs] (group-by :prefixed-queue jobs)]
+    (apply car/rpush queue-key grouped-jobs)))
+
+(defn- enqueue-due-cron-entries
+  [due-cron-entries]
+  (let [jobs-to-enqueue          (map (comp j/from-description :job-description) due-cron-entries)
```

Comment:
Since cron-job is a scheduled job, shall we assoc `:cron-schedule` time to calculate latency & publish that as a stat?

Reply:
Done